src/Yaapii.Atoms/Map/Sorted.cs

```diff
@@ -0,0 +1,178 @@
+                    items.Sort(cmp);
+                    return new Map.Of<Key, Value>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    public sealed class Sorted<Value> : Map.Envelope<Value>
+    {
+        /// <summary>
+        /// Sorts the given map with the default comperator of the key
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs)
+            : this(pairs, Comparer<string>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, Func<KeyValuePair<string, Value>, KeyValuePair<string, Value>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<string, Value>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<string> cmp)
+            : this(pairs, new KeyComparer<string, Value>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<KeyValuePair<string, Value>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<string, Value>>(pairs);
+                    items.Sort(cmp);
+                    return new Map.Of<string, Value>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    public sealed class Sorted : Map.Envelope
+    {
+        /// <summary>
+        /// Sorts the given map with the default comperator of the key
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs)
+            : this(pairs, Comparer<string>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, Func<KeyValuePair<string, string>, KeyValuePair<string, string>, int> compare)
```

Comment:
I think it would be better if the map accepted one comparer (key) or optionally two seperate comparers (key, value).

Reply:
Two seperate comparers will deny implementations that switch between key or value based on conditions.
The user of the constructor of seperate key and value comparers does not know which one is prefered.

For example sorting by value.length and then by key would not be possible.
Or by FirstLetter(Key) and then by value.

Reply:
I think the main usecase is sorting by key. Maybe by value. I do not think we have the need for the complex case you described, and should go for making the main usecase comfortable and work with our lazy map.